src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -32,7 +32,108 @@ GlobalDefaults:
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "AggregationOr"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                {$expr:
+                   {$or:
+                      {^Array: {of: {$and: [
+                        {$eq:
+                           ["$a",
+                            {^RandomInt:
+                               {min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000}}}}]},
+                        {$eq:
+                           ["$b",
+                            {^RandomInt:
+                               {min: 2, max: {^Parameter: { Name: "bMaxValue", Default: 1000}}}}]}]},
+                                number: {^Parameter: { Name: "NumberOfClauses", Default: 50}}}}}}
+
+- TemplateName: MatchOrWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "MatchOr"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                { $or:
+                    { ^Array:
+                        { of:
+                            { a: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000 } } } },
+                              b: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "bMaxValue", Default: 1000 } } } } },
+                          number: { ^Parameter: { Name: "NumberOfClauses", Default: 50 } } } } }
+
+- TemplateName: MatchAndWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "MatchAnd"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {}}}
+
+Array: &Array
+  {^Cycle: {ofLength: 1, fromGenerator: {^Array: {number: 200, of: {^Inc: {start: 0, step: 1}}}}}}
+
+FiftyClauseRootedAnd: &FiftyClauseRootedAnd
```

Comment:
I wasn't able to dedup these three queries; any suggestions on how to use anchors to deduplicate here would be appreciated.

Reply:
Aren't they identical but they have a different array length? Can't the array length just be treated as a parameter?

In fact, can the actual generator defining how to generate the query predicate be moved into the `MatchAndWithManyChildClausesTemplate`, and then have the template accept the array length parameter?

Reply:
Oh dang 🤦  it occurs to me I could the same thing as the or expressions, lol. Done. 

Unlike the $or predicates which have random values, I think I was having trouble generating the fixed sequential values needed for $and.

Reply:
Latest changes here look good to me. Feel free to resolve this thread.